style-guide.md

```diff
@@ -0,0 +1,56 @@
+# WDL Style Guide
+
+All rules below should be followed by contributers to this repo. Pull Requests which do not conform to these specifications will be asked to change.
+
+## Rules
+
+- All WDL should be written in v1.0
+- The following sections must be present and in this order: `meta`, `parameter_meta`, `input`, `command`, `output`, `runtime`
+- The `meta` section should have a `description` of the task
+  - `meta` should *not* include `author` or `email` keys
+- All inputs must have a corresponding `parameter_meta` entry
+  - These texts should be copy and pasted from other tasks with the same input when possible
+    - see `template/common-parameter-meta.txt` for common description strings.
+  - Inputs and parameter meta must be in the same order
+  - That order should be logical
```

Comment:
I think this needs more defining. How would you implement this if you had to write a tool to check for it?

Reply:
I would suggest we be more concrete. For example, input files, output naming, tool arguments, then common arguments, or something similar.

Reply:
I agree this is too vague. I was struggling to come up with concrete rules, but I think Andrew is on the right track. Implementing that check in a linter would be difficult.

Unless we wanted to enforce something like: `File`, `File?`, `String`, `String?`, `Int`, etc. That would be trivial to test in a linter, but I don't think it leads to the best code.

Reply:
It would be really nice if WDL supported semantic tagging of some of these things using keywords. That can get messy really quickly, but even just distinguishing between inputs, outputs, and non-descript parameters would be a pretty good balance.

The other option you have to consider is something to do with alphabetizing things. I don't have a clear answer on this, but would advocate we choose something practical and objective that a tool can achieve, even at the risk of less elegant code. We can always build on it from there.

Reply:
Best compromise between "easy to implement in code" and "elegant looking" that I could come up with:
- Required inputs first
  - next optional
  - last inputs with defaults
- within above groups:
  - group by var type
  - `File` first
  - IDK what the rest of the ordering should be haha

I'm just going to play around with some tasks in the repo and see what looks best to me. But I'd like to avoid alphabetizing (annoying to insert things into the right order) and I think I'm a fan of leaving a little bit of wiggle room to developers. IMO it would be best to leave it up to the developer whether `File bam` or `File fasta` comes first (though I do believe `bam` should come first, but I don't want to get into the business of guessing what the intended uses of the variables are)

Another benefit of leaving *some* ordering up to developers that I'm just thinking of: it will allow you to match the ordering of `foo_tool -h` output. That's obviously an arbitrary ordering per tool, but IMO the order should be reflected in WDL (and there's no way for a linter to know what that ordering should be).